### PR TITLE
Vault Support + Tenant Mode

### PR DESCRIPTION
**Pull request by whitleykeith**
### Description

This PR introduces the support for using Vault as the secrets backend for Airflow instances created from this repo. It also adds the ability to override those variables from within Airflow via the `overrides` variable key. There's some general refactoring to enable the overrides. 

This also adds the "tenant" mode of installation for Airflow within our baremetal cluster. Unlike the playground mode, this only lets a git user deploy 1 instance of Airflow and is not branch-aware. This should be the default method of installation for users looking to have a longer-running instance of Airflow within our cluster. Playgrounds should be reserved specifically for development. This change also introduced some labels to the resources created by these modes to make it easy to prune playgrounds that exist too long. 

More importantly there's a good number of doc changes to make the repo more self-explanatory. 

### Fixes

* es issue where it could only support http:// urls
* issue in airflow where you have to add each module to the path for local imports to work. Resolved by adding the top level package in it's `__init__.py`
* issue where repos cloned with the ssh url don't install properly

adding @kedark3

### CODE REVIEW

**Review by jtaleric — COMMENTED** [review]
I am still not 100% clear on Tenant vs Playground. Other than Playground can go away at any time.

**Comment by whitleykeith** [comment]
Short-term they are definitely very similar, but Playground is mainly meant to be used to test changes before opening PRs against upstream (similar to this one). 

Medium/Long-term we may have cases where a tenant would want to have their own secrets in vault (i.e. QE uses their own AWS creds/jumphost/etc. and they may want to just add their credentials in vault instead of having to override them). We can also pass down the tenant-id to thanos/elasticsearch/dashboard to segment the data.

**Review by jtaleric — APPROVED** [review]
LGTM - thanks for the extra explanation.

**Review by learnitall — APPROVED** [review]
This looks sweet! Love it- thank you for taking the time to do this. Only recommendation is to maybe run black on your Python code and address some unused-imports and f-strings. No big deal, but helps with maintenance over time IMO.

```
[rydrew@redhat-worklp airflow-kubernetes]$ pylint --disable=all --enable unused-import dags/
************* Module dags.openshift_nightlies.dag
dags/openshift_nightlies/dag.py:4:0: W0611: Unused import json (unused-import)
dags/openshift_nightlies/dag.py:5:0: W0611: Unused timedelta imported from datetime (unused-import)
dags/openshift_nightlies/dag.py:5:0: W0611: Unused datetime imported from datetime (unused-import)
dags/openshift_nightlies/dag.py:7:0: W0611: Unused days_ago imported from airflow.utils.dates (unused-import)
dags/openshift_nightlies/dag.py:8:0: W0611: Unused Variable imported from airflow.models (unused-import)
dags/openshift_nightlies/dag.py:10:0: W0611: Unused BashOperator imported from airflow.operators.bash_operator (unused-import)
dags/openshift_nightlies/dag.py:16:0: W0611: Unused BaremetalRelease imported from openshift_nightlies.models.release (unused-import)
dags/openshift_nightlies/dag.py:22:0: W0611: Unused status imported from openshift_nightlies.tasks.index (unused-import)
dags/openshift_nightlies/dag.py:23:0: W0611: Unused var_loader imported from openshift_nightlies.util (unused-import)
************* Module dags.openshift_nightlies.tasks.benchmarks.e2e
dags/openshift_nightlies/tasks/benchmarks/e2e.py:7:0: W0611: Unused import json (unused-import)
dags/openshift_nightlies/tasks/benchmarks/e2e.py:8:0: W0611: Unused timedelta imported from datetime (unused-import)
dags/openshift_nightlies/tasks/benchmarks/e2e.py:10:0: W0611: Unused SubDagOperator imported from airflow.operators.subdag_operator (unused-import)
dags/openshift_nightlies/tasks/benchmarks/e2e.py:11:0: W0611: Unused Variable imported from airflow.models (unused-import)
dags/openshift_nightlies/tasks/benchmarks/e2e.py:12:0: W0611: Unused DAG imported from airflow.models (unused-import)
dags/openshift_nightlies/tasks/benchmarks/e2e.py:14:0: W0611: Unused models imported from kubernetes.client as k8s (unused-import)
************* Module dags.openshift_nightlies.tasks.index.status
dags/openshift_nightlies/tasks/index/status.py:7:0: W0611: Unused import json (unused-import)
dags/openshift_nightlies/tasks/index/status.py:8:0: W0611: Unused import requests (unused-import)
dags/openshift_nightlies/tasks/index/status.py:11:0: W0611: Unused Variable imported from airflow.models (unused-import)
dags/openshift_nightlies/tasks/index/status.py:12:0: W0611: Unused models imported from kubernetes.client as k8s (unused-import)
************* Module dags.openshift_nightlies.tasks.install.openshift
dags/openshift_nightlies/tasks/install/openshift.py:1:0: W0611: Unused import abc (unused-import)
dags/openshift_nightlies/tasks/install/openshift.py:2:0: W0611: Unused constants imported from openshift_nightlies.util (unused-import)
dags/openshift_nightlies/tasks/install/openshift.py:8:0: W0611: Unused import requests (unused-import)
dags/openshift_nightlies/tasks/install/openshift.py:11:0: W0611: Unused BashOperator imported from airflow.operators.bash_operator (unused-import)
dags/openshift_nightlies/tasks/install/openshift.py:12:0: W0611: Unused Variable imported from airflow.models (unused-import)
dags/openshift_nightlies/tasks/install/openshift.py:13:0: W0611: Unused models imported from kubernetes.client as k8s (unused-import)
************* Module dags.openshift_nightlies.tasks.install.baremetal.jetski
dags/openshift_nightlies/tasks/install/baremetal/jetski.py:1:0: W0611: Unused environ imported from os (unused-import)
dags/openshift_nightlies/tasks/install/baremetal/jetski.py:3:0: W0611: Unused executor imported from openshift_nightlies.util (unused-import)
dags/openshift_nightlies/tasks/install/baremetal/jetski.py:11:0: W0611: Unused Variable imported from airflow.models (unused-import)
dags/openshift_nightlies/tasks/install/baremetal/jetski.py:12:0: W0611: Unused models imported from kubernetes.client as k8s (unused-import)
************* Module dags.openshift_nightlies.tasks.install.baremetal.webfuse
dags/openshift_nightlies/tasks/install/baremetal/webfuse.py:1:0: W0611: Unused import sys (unused-import)
dags/openshift_nightlies/tasks/install/baremetal/webfuse.py:2:0: W0611: Unused abspath imported from os.path (unused-import)
dags/openshift_nightlies/tasks/install/baremetal/webfuse.py:2:0: W0611: Unused dirname imported from os.path (unused-import)
dags/openshift_nightlies/tasks/install/baremetal/webfuse.py:5:0: W0611: Unused executor imported from openshift_nightlies.util (unused-import)
dags/openshift_nightlies/tasks/install/baremetal/webfuse.py:12:0: W0611: Unused Variable imported from airflow.models (unused-import)
dags/openshift_nightlies/tasks/install/baremetal/webfuse.py:13:0: W0611: Unused models imported from kubernetes.client as k8s (unused-import)
************* Module dags.openshift_nightlies.tasks.install.cloud.openshift
dags/openshift_nightlies/tasks/install/cloud/openshift.py:1:0: W0611: Unused var_loader imported from openshift_nightlies.util (unused-import)
dags/openshift_nightlies/tasks/install/cloud/openshift.py:1:0: W0611: Unused executor imported from openshift_nightlies.util (unused-import)
dags/openshift_nightlies/tasks/install/cloud/openshift.py:2:0: W0611: Unused StatusIndexer imported from openshift_nightlies.tasks.index.status (unused-import)
dags/openshift_nightlies/tasks/install/cloud/openshift.py:5:0: W0611: Unused import json (unused-import)
dags/openshift_nightlies/tasks/install/cloud/openshift.py:6:0: W0611: Unused import requests (unused-import)
dags/openshift_nightlies/tasks/install/cloud/openshift.py:9:0: W0611: Unused Variable imported from airflow.models (unused-import)
dags/openshift_nightlies/tasks/install/cloud/openshift.py:10:0: W0611: Unused models imported from kubernetes.client as k8s (unused-import)
************* Module dags.openshift_nightlies.tasks.install.openstack.jetpack
dags/openshift_nightlies/tasks/install/openstack/jetpack.py:1:0: W0611: Unused environ imported from os (unused-import)
dags/openshift_nightlies/tasks/install/openstack/jetpack.py:3:0: W0611: Unused var_loader imported from openshift_nightlies.util (unused-import)
dags/openshift_nightlies/tasks/install/openstack/jetpack.py:3:0: W0611: Unused executor imported from openshift_nightlies.util (unused-import)
dags/openshift_nightlies/tasks/install/openstack/jetpack.py:7:0: W0611: Unused import requests (unused-import)
dags/openshift_nightlies/tasks/install/openstack/jetpack.py:10:0: W0611: Unused Variable imported from airflow.models (unused-import)
dags/openshift_nightlies/tasks/install/openstack/jetpack.py:11:0: W0611: Unused models imported from kubernetes.client as k8s (unused-import)
************* Module dags.openshift_nightlies.tasks.utils.scale_ci_diagnosis
dags/openshift_nightlies/tasks/utils/scale_ci_diagnosis.py:3:0: W0611: Unused StatusIndexer imported from openshift_nightlies.tasks.index.status (unused-import)
dags/openshift_nightlies/tasks/utils/scale_ci_diagnosis.py:6:0: W0611: Unused import json (unused-import)
dags/openshift_nightlies/tasks/utils/scale_ci_diagnosis.py:7:0: W0611: Unused timedelta imported from datetime (unused-import)
dags/openshift_nightlies/tasks/utils/scale_ci_diagnosis.py:9:0: W0611: Unused SubDagOperator imported from airflow.operators.subdag_operator (unused-import)
dags/openshift_nightlies/tasks/utils/scale_ci_diagnosis.py:10:0: W0611: Unused Variable imported from airflow.models (unused-import)
dags/openshift_nightlies/tasks/utils/scale_ci_diagnosis.py:11:0: W0611: Unused DAG imported from airflow.models (unused-import)
dags/openshift_nightlies/tasks/utils/scale_ci_diagnosis.py:12:0: W0611: Unused TaskGroup imported from airflow.utils.task_group (unused-import)
dags/openshift_nightlies/tasks/utils/scale_ci_diagnosis.py:13:0: W0611: Unused models imported from kubernetes.client as k8s (unused-import)
************* Module dags.openshift_nightlies.util.var_loader
dags/openshift_nightlies/util/var_loader.py:3:0: W0611: Unused executor imported from openshift_nightlies.util (unused-import)
dags/openshift_nightlies/util/var_loader.py:6:0: W0611: Unused import requests (unused-import)
dags/openshift_nightlies/util/var_loader.py:8:0: W0611: Unused models imported from kubernetes.client as k8s (unused-import)
```

File: README.md [hunk 1]
```diff
@@ -1,42 +1,39 @@
...
+
+To install Airflow as a Tenant, you can run the following commands from the fork you wish to use as your tenant repo.
 
 ```bash
 # all commands are run at the root of your git repo
 # install the airflow stack and have it point to your fork of the dag code.
-# $PASSWORD refers to the password you would like to secure your airflow instance behind.
-./scripts/install.sh -p $PASSWORD
+# $PASSWORD refers to the password you would like to secure your airflow instance with.
+./scripts/tenant/create.sh -p $PASSWORD
```
Comment (kedark3) [inline README.md:26]: for the sake of consistency with Playground, why don't we call these `build.sh` and `cleanup.sh` ? `create` and `destroy` are fine but lack consistency.  @whitleykeith ?

File: README.md [hunk 2]
```diff
@@ -47,5 +44,26 @@ To install Airflow in a developer playground setting (i.e. in our baremetal clus
...
 
-## Cleaning up the Playground
-To uninstall the stack, you can run `./scripts/playground/cleanup.sh`.
\ No newline at end of file
+### Cleaning up the Playground
+To uninstall the stack, you can run `./scripts/playground/cleanup.sh`.
+
+---
+
+## Installing the PerfScale Stack
```
Comment (kedark3) [inline README.md:52]: Can you define (in just 1 line ) what all is included in PerfScale Stack ?

File: dags/openshift_nightlies/docs/secrets.md [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+    "elasticsearch": "my_custom_elastic_url"
+}
+
+```
+
+
+
+
+
+# Supported Variables
```
Comment (kedark3) [inline dags/openshift_nightlies/docs/secrets.md:39]: Its a good idea to list vars here, but keeping it up-to-date will be a challenge :P Let's keep an eye on this list.

File: README.md (hunk 1 above)
Comment (whitleykeith) [reply README.md:26]: the main reason for naming them differently is to ensure people don't accidentally call the wrong script by using the wrong directory (which has been attempted multiple times with the standalone and playground modes).

medium term I want to add commands to the makefile to where someone does `make build playground` which will make the problem less persistent. At that point I'm not against making them the same name

File: dags/openshift_nightlies/docs/secrets.md (hunk 3 above)
Comment (whitleykeith) [reply dags/openshift_nightlies/docs/secrets.md:39]: yeah I plan to block PRs if they add a new variable w/o updating the doc :D

File: README.md (hunk 2 above)
Comment (whitleykeith) [reply README.md:52]: Yeah I'll add that change here https://github.com/whitleykeith/airflow-kubernetes/tree/ci

**Pull request merged**